website/routes.py

```diff
@@ -765,6 +765,12 @@ def make_url_map(app):
         ], 'get', project_views.node.node_registrations,
             OsfWebRenderer('project/registrations.mako')),
 
+        Rule([
+            '/project/<pid>/retract_registration/',
```

Comment:
@brianjgeiger Valid point. There are clearly multiple approaches. GitHub, for example, allows you to star with `PUT /gists/:id/star` and unstar with `DELETE /gists/:id/star` (though this is still ambiguous, since "star" can be either a noun or a verb).

I think `/project/<pid>/retraction/` is sensible. Let's go with that.